scripts/gen_vimdoc.py

```diff
@@ -187,6 +187,23 @@
         'module_override': {},
         'append_only': [],
     },
+    'diagnostic': {
+        'mode': 'lua',
+        'filename': 'diagnostic.txt',
+        'section_start_token': '*diagnostic-api*',
+        'section_order': [
+            'diagnostic.lua',
+        ],
+        'files': os.path.join(base_dir, 'runtime/lua/vim/diagnostic.lua'),
+        'file_patterns': '*.lua',
+        'fn_name_prefix': '',
+        'section_name': {'diagnostic.lua': 'diagnostic'},
+        'section_fmt': lambda _: 'DIAGNOSTIC API',
```

Comment:
convention of lua, lsp, treesitter is:

```suggestion
        'section_fmt': lambda _: 'Lua module: vim.diagnostic',
```